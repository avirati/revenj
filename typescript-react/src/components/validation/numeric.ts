import {
  Num,
  ValidatorCreator,
} from './interfaces';
import * as operators from './operators';
import { validatorCreatorFactory } from './validatorCreatorFactory';

////////////////////////////////////////////////////////////////////////////////
// Order validators                                                           //
////////////////////////////////////////////////////////////////////////////////
interface INumValidatorCreatorFactory {
  (value: Num): ValidatorCreator<Num, Num, any, any>;
}

export const multipleOfFactorCreator: INumValidatorCreatorFactory = (value: Num) =>
  validatorCreatorFactory<Num, Num, any, any>({
    getValidatorBaseValue: () => value,
    getValidatorErrorMessage: (base) => `Value must be a multiple of ${base}.`,
    operator: operators.multipleOfFactor,
  });
