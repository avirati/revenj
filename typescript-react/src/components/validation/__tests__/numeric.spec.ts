import { BigNumber } from 'bignumber.js';

import { multipleOfFactorCreator } from '../numeric';

describe('validation', () => {
  describe('multipleOfFactorCreator', () => {
    const multipleOf7 = multipleOfFactorCreator(7)();
    const multipleOfBn7 = multipleOfFactorCreator(new BigNumber(7))();

    it('should return undefined if input value is a multiple of base value', () => {
      expect(multipleOf7(14, {})).toBeUndefined();
      expect(multipleOfBn7(new BigNumber(14), {})).toBeUndefined();
    });

    it('should return an error message if input value is not a multiple of base value', () => {
      expect(multipleOf7(31, {})).toBeDefined();
      expect(multipleOf7(31, {})).toMatchSnapshot();
      expect(multipleOfBn7(new BigNumber(-3), {})).toBeDefined();
      expect(multipleOfBn7(new BigNumber(-3), {})).toMatchSnapshot();
    });
  });
});
