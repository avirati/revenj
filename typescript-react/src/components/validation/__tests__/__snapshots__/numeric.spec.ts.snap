// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation multipleOfFactorCreator should return an error message if input value is not a multiple of base value 1`] = `"Value must be a multiple of 7."`;

exports[`validation multipleOfFactorCreator should return an error message if input value is not a multiple of base value 2`] = `"Value must be a multiple of 7."`;
